### PR TITLE
adding g2cf.F90

### PR DESCRIPTION
**Pull request by edwardhartnett**
Part of #328

### CODE REVIEW

**Comment by edwardhartnett** [comment]
OK, this must wait for #708 to be merged before it can proceed.

File: src/g2c_interface.F90 [hunk 1]
```diff
@@ -0,0 +1,36 @@
+! Contains a module with interfaces to the C functions in the g2c
+! library.
+!
+! Note: this file is excluded from the doxygen build (see Doxyfile.in)
+! because these functions are already documented in the g2c library.
```
Comment (AlexanderRichert-NOAA) [inline src/g2c_interface.F90:5]: Does/should the doxygen docs have some indication of which functions have Fortran interfaces?

File: src/g2cf.F90 [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+       cstring = string(1:nlen)
+    else ! append null char to trimmed string
+       cstring = string(1:nlen)//C_NULL_CHAR
+       nlen = nlen + 1
+    endif
+  end function addcnullchar
+
+  !> Open a GRIB2 file.
+  !>
+  !> @param path the path to the file
```
Comment (AlexanderRichert-NOAA) [inline src/g2cf.F90]: Did we decide to capitalize the doxygen descriptions of variables? Personally I would find it easier to read if either the first letter of descriptions were upper case, or if there were some separator like a hyphen.

File: src/g2cf.F90 [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    integer :: ie1, ie2
+    
+    cmode = mode
+    cg2cid = 0
+    
+    ! check for c null character on path and add one if not present.
+    cdata_file = addcnullchar(data_file, ie1)
+    cindex_file = addcnullchar(index_file, ie2)
+    
+    ! call nc_create to create file
```
Comment (AlexanderRichert-NOAA) [inline src/g2cf.F90]: I'm assuming you don't mean nc_create :)

File: src/g2c_interface.F90 (hunk 1 above)
Comment (edwardhartnett) [reply src/g2c_interface.F90:5]: No users access everything through the fortran API and never call these functions directly.

File: src/g2cf.F90 (hunk 2 above)
Comment (edwardhartnett) [reply src/g2cf.F90]: Fixed, thanks.

**Review by Hang-Lei-NOAA — APPROVED** [review]

**Pull request merged**